examples/light-switch-app/nrfconnect/main/LightSwitch.cpp

```diff
@@ -75,7 +76,8 @@ void LightSwitch::DimmerChangeBrightness()
         {
             brightness = 0;
         }
-        data->Value = (uint8_t) brightness;
+        data->Value   = (uint8_t) brightness;
```

Comment:
Why is the cast needed here? Or why is the brightness defined as `uint16_t`?

Reply:
It's defined as uint16_t because maximum brightness equals 254, and I approximate changing brightness by 1% step as brightness += 3. If the brightness variable were uint8_t type, It would have been overflowed.

Reply:
But isn't it that if you let's say add 3 to 254 (so you have 257) which on 2 byte variable is encoded as 0x0101, and if you now cast it to uint8_t you will not have 255 but simply the MSB will be stripped and you and end up with `data->Value` equal to 1? For me it looks like we should rather have a condition protecting the brightness going beyond 255 (or 254).